liquidsoap.spec

```diff
@@ -72,9 +97,9 @@ components working together.
 make
 
 %install
-make install DESTDIR=%{buildroot}%{_exec_prefix} OCAMLFIND_DESTDIR=%{buildroot}%{_exec_prefix} prefix=%{buildroot}%{_exec_prefix} sysconfdir=%{buildroot}/etc mandir=%{buildroot}%{_exec_prefix}/share/man localstatedir=%{buildroot}/var
-/bin/install -d %{buildroot}%{_exec_prefix}/lib/systemd/system/
-/bin/install -c %{SOURCE1} -m 644 %{buildroot}%{_exec_prefix}/lib/systemd/system/
+%make_install%{_exec_prefix} OCAMLFIND_DESTDIR=%{buildroot}%{_exec_prefix} prefix=%{buildroot}%{_exec_prefix} sysconfdir=%{buildroot}/etc mandir=%{buildroot}%{_exec_prefix}/share/man localstatedir=%{buildroot}/var
```

Comment:
Is this actually working/correct? `%make_install%{_exec_prefix}`

Are all those directory variable specifications really necessary, if you specify `prefix=%{buildroot}%{_exec_prefix}`? If yes, you might want to use the [autoconf derived macros](https://fedoraproject.org/wiki/Packaging:RPMMacros#Macros_mimicking_autoconf_variables) for those directories (such as` sysconfdir=%{_sysconfdir}` `mandir=%{_mandir}` etc.), rather than building them by yourself.

Reply:
```
$ rpmbuild -E '%make_install'
/usr/bin/make install DESTDIR=/root/rpmbuild/BUILDROOT/%{name}-%{version}-%{release}.x86_64
$ rpmbuild -E '%make_install%{_exec_prefix}'
/usr/bin/make install DESTDIR=/root/rpmbuild/BUILDROOT/%{name}-%{version}-%{release}.x86_64/usr
```

Reply:
This additional diff
```diff
--- a/liquidsoap.spec
+++ b/liquidsoap.spec
@@ -97,7 +97,7 @@ components working together.
 make
 
 %install
-%make_install%{_exec_prefix} OCAMLFIND_DESTDIR=%{buildroot}%{_exec_prefix} prefix=%{buildroot}%{_exec_prefix} sysconfdir=%{buildroot}/etc mandir=%{buildroot}%{_exec_prefix}/share/man localstatedir=%{buildroot}/var
+%make_install%{_exec_prefix} OCAMLFIND_DESTDIR=%{buildroot}%{_exec_prefix} prefix=%{buildroot}%{_exec_prefix} sysconfdir=%{_sysconfdir} mandir=%{_mandir} localstatedir=%{_localstatedir}
 /bin/install -d %{buildroot}%{_unitdir}
 /bin/install -c %{SOURCE1} -m 644 %{buildroot}%{_unitdir}
 
```
leads to 
```
RPM build errors:
    File not found: /root/rpmbuild/BUILDROOT/liquidsoap-1.3.3-1.el7.centos.x86_64/etc/liquidsoap/radio.liq.example
    File not found: /root/rpmbuild/BUILDROOT/liquidsoap-1.3.3-1.el7.centos.x86_64/etc/logrotate.d/liquidsoap
    File not found by glob: /root/rpmbuild/BUILDROOT/liquidsoap-1.3.3-1.el7.centos.x86_64/usr/share/man/man1/liquidsoap.1.*
```
All of this is somewhat specialised for the OCaml case hence autoconf macros not really working for all of it. I plan on addressing what remains once the OCaml build situation trickles down to CentOS since it's already a bit cleaner in Fedora.

Reply:
Hmm... can't you use the normal `%configure` and `%make_install` macros here, which would already use the correct directory paths? Something like:

```
%build
%configure --with-internal-glib --disable-camomile
make %{?_smp_mflags}

%install
%make_install OCAMLFIND_DESTDIR=...
/bin/install -d %{buildroot}%{_unitdir}
/bin/install -c %{SOURCE1} -m 644 %{buildroot}%{_unitdir}
```

Of course it might require some additional tweaking.

Anyway, you probably already tried that, let me know if you want to merge the current state.

Reply:
Using `%configure` as per your example leads errors as well.

As we discussed, let's merge this as is and take care of a larger cleanup (also concerning all the other centos-rpm-ocaml-* repos) at a later stage when the Fedora packaging guidelines for OCaml are more stable.

I expect a reasonable timeframe for this to be sometime after the release of CentOS-8 which also coincides with me wanting to start upstreaming this to Fedora rawhide sometime after that milestone.